### ISSUE TITLE
Can't seem to find job results using docker-compose

### ISSUE THREAD

**Issue by mennymendoza** [TheHassi/superset-pdf-report#1]
So I followed [this guide](https://superset.apache.org/docs/installation/installing-superset-using-docker-compose) for installing Apache Superset using Docker Compose. I'm currently running it in non-dev mode on an Ubuntu VM with no issues. I've also successfully configured thumbnail caching with the following superset_config.py configuration:

```
import logging
import os
from datetime import timedelta
from typing import Optional

from cachelib.file import FileSystemCache
from celery.schedules import crontab
from superset.superset_typing import CacheConfig

logger = logging.getLogger()


def get_env_variable(var_name: str, default: Optional[str] = None) -> str:
    """Get the environment variable or raise exception."""
    try:
        return os.environ[var_name]
    except KeyError:
        if default is not None:
            return default
        else:
            error_msg = "The environment variable {} was missing, abort...".format(
                var_name
            )
            raise EnvironmentError(error_msg)


DATABASE_DIALECT = get_env_variable("DATABASE_DIALECT")
DATABASE_USER = get_env_variable("DATABASE_USER")
DATABASE_PASSWORD = get_env_variable("DATABASE_PASSWORD")
DATABASE_HOST = get_env_variable("DATABASE_HOST")
DATABASE_PORT = get_env_variable("DATABASE_PORT")
DATABASE_DB = get_env_variable("DATABASE_DB")

# The SQLAlchemy connection string.
SQLALCHEMY_DATABASE_URI = "%s://%s:%s@%s:%s/%s" % (
    DATABASE_DIALECT,
    DATABASE_USER,
    DATABASE_PASSWORD,
    DATABASE_HOST,
    DATABASE_PORT,
    DATABASE_DB,
)

REDIS_HOST = get_env_variable("REDIS_HOST")
REDIS_PORT = get_env_variable("REDIS_PORT")
REDIS_CELERY_DB = get_env_variable("REDIS_CELERY_DB", "0")
REDIS_RESULTS_DB = get_env_variable("REDIS_RESULTS_DB", "1")

BASE_CACHE_DIR = "/app/superset_home/sqllab"
CACHE_THRESHOLD = 1000
CACHE_DEFAULT_TIMEOUT = 100

FEATURE_FLAGS = {"ALERT_REPORTS": True, "THUMBNAILS": True, "THUMBNAILS_SQLA_LISTENERS": True}
CACHE_CONFIG: CacheConfig = {
    'CACHE_TYPE': 'redis',
    'CACHE_DEFAULT_TIMEOUT': 24*60*60, # 1 day
    'CACHE_KEY_PREFIX': 'superset_',
    'CACHE_REDIS_URL': 'redis://superset_cache:6379/3'
}
DATA_CACHE_CONFIG: CacheConfig = {
    'CACHE_TYPE': 'redis',
    'CACHE_DEFAULT_TIMEOUT': 24*60*60, # 1 day
    'CACHE_KEY_PREFIX': 'data_',
    'CACHE_REDIS_URL': 'redis://superset_cache:6379/3'
}
THUMBNAIL_SELENIUM_USER = "admin"
THUMBNAIL_CACHE_CONFIG: CacheConfig = {
    'CACHE_TYPE': 'redis',
    'CACHE_DEFAULT_TIMEOUT': 24*60*60,
    'CACHE_KEY_PREFIX': 'thumbnail_',
    'CACHE_NO_NULL_WARNING': True,
    'CACHE_REDIS_URL': 'redis://superset_cache:6379/3'
}

RESULTS_BACKEND = FileSystemCache(
    cache_dir=os.path.join(BASE_CACHE_DIR, 'results/'),
    threshold=CACHE_THRESHOLD,
    default_timeout=CACHE_DEFAULT_TIMEOUT
)

SCREENSHOT_LOCATE_WAIT = 100
SCREENSHOT_LOAD_WAIT = 600

class CeleryConfig(object):
    BROKER_URL = f"redis://{REDIS_HOST}:{REDIS_PORT}/{REDIS_CELERY_DB}"
    CELERY_IMPORTS = ("superset.sql_lab", "superset.tasks", 'superset.tasks.thumbnails', "supersetpdfreport.task")
    CELERY_RESULT_BACKEND = f"redis://{REDIS_HOST}:{REDIS_PORT}/{REDIS_RESULTS_DB}"
    CELERYD_LOG_LEVEL = "DEBUG"
    CELERYD_PREFETCH_MULTIPLIER = 10
    CELERY_ACKS_LATE = True
    CELERY_ANNOTATIONS = {
        'sql_lab.get_sql_results': {
            'rate_limit': '100/s',
        },
        'email_reports.send': {
            'rate_limit': '1/s',
            'time_limit': 600,
            'soft_time_limit': 600,
            'ignore_result': True,
        },
    }
    CELERYBEAT_SCHEDULE = {
        'reports.scheduler': {
            'task': 'reports.scheduler',
            'schedule': crontab(minute='*', hour='*'),
        },
        'reports.prune_log': {
            'task': 'reports.prune_log',
            'schedule': crontab(minute=0, hour=0),
        },
        'cache-warmup-hourly': {
            'task': 'cache-warmup',
            'schedule': crontab(minute='*/30', hour='*'),
            'kwargs': {
                'strategy_name': 'top_n_dashboards',
                'top_n': 10,
                'since': '7 days ago',
            },
        },
        'reports.pdf-scheduler': {
            "task": "reports.pdf-scheduler",
            "schedule": crontab(minute="*/1")
        }
    }


CELERY_CONFIG = CeleryConfig

ALERT_REPORTS_NOTIFICATION_DRY_RUN = True

WEBDRIVER_TYPE = "firefox"
WEBDRIVER_BASEURL = "http://172.17.0.1:8088"
WEBDRIVER_BASEURL_USER_FRIENDLY = WEBDRIVER_BASEURL


SQLLAB_CTAS_NO_LIMIT = True

try:
    import superset_config_docker
    from superset_config_docker import *  # noqa

    logger.info(
        f"Loaded your Docker configuration at " f"[{superset_config_docker.__file__}]"
    )
except ImportError:
    logger.info("Using default Docker config...")
```

Everything seems to be completely fine and I am able to use the Superset API to cache and get thumbnails. I've added the pdf-scheduler to my CELERYBEAT_SCHEDULE and have added superset-pdf-report to my requirements-local.txt so I get no errors in the logs. I've added the following environment variables to my .env-non-dev file:

```
# Folder for the log files
LOG_FOLDER="/app/superset-pdf/sspdf-logs/"
# Folder for the jobs
PDF_REPORT_JOB_FOLDER="/app/superset-pdf/sspdf-jobs"
# Configuration for Superset-API
WEBSERVER_PROTOCOL="http"
WEBSERVER_ADDRESS="172.17.0.1"
WEBSERVER_PORT="8088"
SUPERSET_API_USER="admin"
SUPERSET_API_USER_FILE="/app/superset-pdf/userfile.json"
SUPERSET_API_PASSWORD="admin"
SUPERSET_API_PASSWORD_FILE="/app/superset-pdf/userfile.json"

# Configuration for E-Mail Account
SMTP_HOST="127.0.0.1"
SMTP_PORT="25"
SMTP_USER="admin"
SMTP_PASSWORD="admin"
SMTP_PASSWORD_FILE="/app/superset-pdf/userfile.json"
SMTP_MAIL_FROM="admin"
SMTP_STARTTLS="admin"
SMTP_SSL="admin"

# Configuration for nextcloud webdav
NEXTCLOUD_URL="http://127.0.0.1:8000" # this is just some garbage since i don't plan to use nextcloud or SMTP
NEXTCLOUD_USER="admin"
NEXTCLOUD_PASSWORD="admin"
NEXTCLOUD_USER_FILE="/app/superset-pdf/userfile.json"
NEXTCLOUD_PASSWORD_FILE="/app/superset-pdf/userfile.json"
```

However, I'm not sure where the output is supposed to go? I've looked in the superset_app container and I see no change. Here are some snippets of the logs for superset_worker, superset_worker_beat and superset_cache.

superset_worker
```
[2022-06-29 23:26:27,221: INFO/MainProcess] Task cache_chart_thumbnail[54887f3d-c5a5-4652-b326-3562fb8e5e38] received
Caching chart: http://172.17.0.1:8088/superset/slice/1/?standalone=true
[2022-06-29 23:26:27,223: INFO/ForkPoolWorker-1] Caching chart: http://172.17.0.1:8088/superset/slice/1/?standalone=true
Processing url for thumbnail: 04b93bebfa2590f8e8b4a52ef160a50a
[2022-06-29 23:26:27,244: INFO/ForkPoolWorker-1] Processing url for thumbnail: 04b93bebfa2590f8e8b4a52ef160a50a
Init selenium driver
[2022-06-29 23:26:27,246: INFO/ForkPoolWorker-1] Init selenium driver
Taking a PNG screenshot of url http://172.17.0.1:8088/superset/slice/1/?standalone=true
[2022-06-29 23:26:45,327: INFO/ForkPoolWorker-1] Taking a PNG screenshot of url http://172.17.0.1:8088/superset/slice/1/?standalone=true
Caching thumbnail: 04b93bebfa2590f8e8b4a52ef160a50a
[2022-06-29 23:26:45,891: INFO/ForkPoolWorker-1] Caching thumbnail: 04b93bebfa2590f8e8b4a52ef160a50a
Done caching thumbnail
[2022-06-29 23:26:45,895: INFO/ForkPoolWorker-1] Done caching thumbnail
[2022-06-29 23:26:45,905: INFO/ForkPoolWorker-1] Task cache_chart_thumbnail[54887f3d-c5a5-4652-b326-3562fb8e5e38] succeeded in 18.683201605999784s: None
```

superset_worker_beat
```
[2022-06-29 23:59:00,038: INFO/MainProcess] Scheduler: Sending due task reports.pdf-scheduler (reports.pdf-scheduler)
[2022-06-30 00:00:00,048: INFO/MainProcess] Scheduler: Sending due task reports.prune_log (reports.prune_log)
[2022-06-30 00:00:00,053: INFO/MainProcess] Scheduler: Sending due task cache-warmup-hourly (cache-warmup)
[2022-06-30 00:00:00,055: INFO/MainProcess] Scheduler: Sending due task reports.scheduler (reports.scheduler)
[2022-06-30 00:00:00,059: INFO/MainProcess] Scheduler: Sending due task reports.pdf-scheduler (reports.pdf-scheduler)
[2022-06-30 00:01:00,047: INFO/MainProcess] Scheduler: Sending due task reports.pdf-scheduler (reports.pdf-scheduler)
[2022-06-30 00:01:00,050: INFO/MainProcess] Scheduler: Sending due task reports.scheduler (reports.scheduler)
[2022-06-30 00:02:00,051: INFO/MainProcess] Scheduler: Sending due task reports.pdf-scheduler (reports.pdf-scheduler)
[2022-06-30 00:02:00,053: INFO/MainProcess] Scheduler: Sending due task reports.scheduler (reports.scheduler)
[2022-06-30 00:03:00,050: INFO/MainProcess] Scheduler: Sending due task reports.pdf-scheduler (reports.pdf-scheduler)
[2022-06-30 00:03:00,052: INFO/MainProcess] Scheduler: Sending due task reports.scheduler (reports.scheduler)
[2022-06-30 00:04:00,049: INFO/MainProcess] Scheduler: Sending due task reports.pdf-scheduler (reports.pdf-scheduler)
[2022-06-30 00:04:00,052: INFO/MainProcess] Scheduler: Sending due task reports.scheduler (reports.scheduler)
```

superset_cache
```
46:C 29 Jun 2022 23:44:00.091 * Fork CoW for RDB: current 0 MB, peak 0 MB, average 0 MB
1:M 29 Jun 2022 23:44:00.195 * Background saving terminated with success
1:M 29 Jun 2022 23:52:00.114 * 100 changes in 300 seconds. Saving...
1:M 29 Jun 2022 23:52:00.114 * Background saving started by pid 47
47:C 29 Jun 2022 23:52:00.118 * DB saved on disk
47:C 29 Jun 2022 23:52:00.119 * Fork CoW for RDB: current 0 MB, peak 0 MB, average 0 MB
1:M 29 Jun 2022 23:52:00.215 * Background saving terminated with success
1:M 30 Jun 2022 00:00:00.141 * 100 changes in 300 seconds. Saving...
1:M 30 Jun 2022 00:00:00.142 * Background saving started by pid 48
48:C 30 Jun 2022 00:00:00.144 * DB saved on disk
48:C 30 Jun 2022 00:00:00.144 * Fork CoW for RDB: current 0 MB, peak 0 MB, average 0 MB
1:M 30 Jun 2022 00:00:00.242 * Background saving terminated with success
```

The superset_app container doesn't give me any indication it is continuously breaking (since superset_worker_beat would have me believe it's continuously executing the job). But here is something weird that I found in the logs. For the most part though, superset_app seems to be working fine.
```
172.22.0.1 - - [29/Jun/2022:23:41:10 +0000] "GET /api/v1/chart/cache_screenshot HTTP/1.1" 405 271 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:102.0) Gecko/20100101 Firefox/102.0"
HTTPException
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1492, in dispatch_request
    self.raise_routing_exception(req)
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1474, in raise_routing_exception
    raise request.routing_exception  # type: ignore
  File "/usr/local/lib/python3.8/site-packages/flask/ctx.py", line 376, in match_request
    result = self.url_adapter.match(return_rule=True)  # type: ignore
  File "/usr/local/lib/python3.8/site-packages/werkzeug/routing.py", line 2037, in match
    raise MethodNotAllowed(valid_methods=list(have_match_for))
werkzeug.exceptions.MethodNotAllowed: 405 Method Not Allowed: The method is not allowed for the requested URL.
2022-06-29 23:41:10,896:WARNING:superset.views.base:HTTPException
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1492, in dispatch_request
    self.raise_routing_exception(req)
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1474, in raise_routing_exception
    raise request.routing_exception  # type: ignore
  File "/usr/local/lib/python3.8/site-packages/flask/ctx.py", line 376, in match_request
    result = self.url_adapter.match(return_rule=True)  # type: ignore
  File "/usr/local/lib/python3.8/site-packages/werkzeug/routing.py", line 2037, in match
    raise MethodNotAllowed(valid_methods=list(have_match_for))
werkzeug.exceptions.MethodNotAllowed: 405 Method Not Allowed: The method is not allowed for the requested URL.
Triggering screenshot ASYNC
2022-06-29 23:41:19,755:INFO:superset.charts.api:Triggering screenshot ASYNC
```

My Docker Compose file is basically exactly the same as the one in the guide. I'm using docker-compose-non-dev.yml.

TLDR; not sure where to find the jobs or where to find any indication that superset-pdf-report is even doing anything. The superset-pdf-report module isn't generating any logs and when I check the superset_worker logs, everything APPEARS to be running smoothly.

**Comment by mennymendoza** [comment]
I ended up rebuilding this tool without the pdflatex dependency, instead using [PyFPDF](https://pyfpdf.readthedocs.io/en/latest/) to push each screenshot to a PDF.

**Comment by ronyrv13** [comment]
Hello @mennymendoza / @TheHassi 
I am noob in apache superset, but i got a requirement of this.
Can you please share that, how this pdf module actually works?
Do this module enables an option to export dashboard/chart on pdf format at UI? ( probably not)
just a brief will be really helpful.
The document does not include much explanation!.

**Comment by mennymendoza** [comment]
With some configuration, you might be able to add a button to the UI, but I'm honestly not 100% positive how that would work. I think I remember seeing something about adding UI buttons in the superset [config.py](https://github.com/apache/superset/blob/master/superset/config.py) file.

However, what I did was rebuild this as a command line tool, where you can run it and it will take all charts and thumbnails and push them to a pdf document. If you look through the module code, you can see that all it's really doing is making API calls through HTTP requests to the [superset backend API](https://superset.apache.org/docs/api) and downloading the charts and thumbnails that way. Once they're downloaded, you can then push them wherever you want; in this module, @TheHassi pushes them to a LaTeX document whereas I simply pushed them to a PDF using PyFPDF. For my use case, I then used the Celery scheduler to schedule regular PDF downloads. However, I personally think that unless you want JUST the individual charts, this method is inferior to simply taking your own screenshots using something like [Puppeteer](https://developer.chrome.com/docs/puppeteer/). I talk about it in [this](https://github.com/apache/superset/issues/20756) issue.
